agent/testflinger_agent/client.py

```diff
@@ -429,6 +429,5 @@ def post_status_update(
             job_request = None
         if not job_request:
             logger.error(
-                "Unable to post status updates to: %s (error: %s)"
-                % (status_update_uri, job_request.status_code)
+                f"Unable to post status updates to: {status_update_uri}"
```

Comment:
There are some exceptions, but in general, it's usually a bad habit to use f-strings in logging messages. Also, this dropped the detail about the status code from the error - any reason for that?

Reply:
I think this should be in a different PR - but I caught a bug here where if there is a RequestException and job_request is None, that status_code doesn't exist.

Reply:
Interesting, do you know how to reproduce it? I thought it would always return a response object, even on error conditions, but perhaps RequestException is not like that. Feel free to submit a separate PR for this if you have a quick way to test that.